src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -264,11 +264,22 @@ public IAsyncCursor<TResult> Execute(IReadBinding binding, CancellationToken can
                 throw new ArgumentException("The binding value passed to ChangeStreamOperation.Execute must implement IReadBindingHandle.", nameof(binding));
             }
 
-            var cursor = Resume(bindingHandle, cancellationToken);
-            if (_startAtOperationTime == null && _resumeAfter == null)
+            IAsyncCursor<RawBsonDocument> cursor;
+            using (var channelSource = binding.GetReadChannelSource(cancellationToken))
+            using (var channel = channelSource.GetChannel(cancellationToken))
+            using (var channelBinding = new ChannelReadBinding(channelSource.Server, channel, binding.ReadPreference, binding.Session.Fork()))
             {
-                _startAtOperationTime = binding.Session.OperationTime;
+                cursor = Resume(channelBinding, cancellationToken);
+                if (_startAtOperationTime == null && _resumeAfter == null)
+                {
+                    var maxWireVersion = channel.ConnectionDescription.IsMasterResult.MaxWireVersion;
+                    if (maxWireVersion >= 7)
+                    {
+                        _startAtOperationTime = binding.Session.OperationTime;
```

Comment:
So this is following spec, right?  If there's a resumable error and we resume on a 3.6 server, it will result in an error?

Reply:
I suppose so. The spec doesn't seem to take the possibility of resuming on a downgraded server into account. Does it even matter? How likely/common is such a scenario?